### DISCUSSION TITLE
Add a HUG feature for the controls, including page.

### DISCUSSION THREAD

**Discussion by elsanguchemecatronico** [flet-dev/flet#2069] [Ideas]
I apologize if this feature is already implemented, but I can't find it.
It would be nice to allow the **layout** controls to **hug** the controls inside of it, just like **auto layout** in Figma works.

In particular, I'm interested in a vertical hug feature for the page control since I have no clue on how to calculate the total height of the controls inside of it (most **height** properties are **none**).

**Comment by ndonkoHenri** [comment]
Please can you provide a link to a clearer explanation  'hug' feature? 

> I have no clue on how to calculate the total height of the controls inside of it (most height properties are none).

I wrote an article on that recently: https://python.plainenglish.io/how-to-determine-the-size-of-any-flet-control-python-ddc5631dd442

**Comment by elsanguchemecatronico** [comment]
Basically, what `hug` would do, is to resize the control to smallest possible dimensions to surround the objects within it.
It's the oposite of `expand`. But while `expand` applies to the children, `hug` applies to the parent.

In my particular case, I will use it for the `page` control, in the height. 
![imagen](https://github.com/flet-dev/flet/assets/147406455/8057c373-ce3c-4752-9e1f-592641fd8f91)

The basic code for the previous GUI is:

```
	import flet as ft

	def main(page: ft.Page):
		txt_issue = ft.TextField(
			label = 'Issue',
			expand = True
			)

		txt_time = ft.TextField(
			label = 'Time Spent',
			expand = False,
			width = 150
			)

		txt_comment = ft.TextField(
			label = 'Comment',
			)

		txt_dates = ft.TextField(
			label = 'Dates',
			#error_text = 'Error' # Uncomment this line to show the "error" message.
			)

		btn_upload = ft.ElevatedButton(
			text = 'Upload Work Log',
			icon = 'upload'
			)

		row_buttons = ft.Row(
			controls = [btn_upload],
			alignment = ft.MainAxisAlignment.CENTER
			)

		row = ft.Row(
			controls = [txt_issue,txt_time]
			)

		col = ft.Column(
			controls = [row,txt_comment,txt_dates,row_buttons],
			expand = True
			)

		st = ft.Stack(
			controls = [col])

		cnt = ft.Container(
			content = st,
			)

		page.add(cnt)

		# Page properties.
		page.window_resizable = True
		page.window_width = 400
		page.window_height = 300 
		page.window_maximizable = False

		page.update()

	ft.app(target = main)
```
The page height was chosen to best fit to its content, but when the height of the dates text field change, I have to change the height of the page manually.

Is it necessary to create a completely new class to know the size of a control?
The same thing happens with the `color` property: since its controlled by the theme, when you read it, you get a `None` value.

**Reply by ndonkoHenri** [reply]
> Is it necessary to create a completely new class to know the size of a control?

Can [this](https://python.plainenglish.io/how-to-determine-the-size-of-any-flet-control-python-ddc5631dd442) be of help?

**Reply by elsanguchemecatronico** [reply]
Thats why I asked if it is necessaary to create a new class just to access the size of the control.

**Reply by ndonkoHenri** [reply]
If the size of a control is not explicitly set, then flet will return None when you try to retrieve its size (width, height).
If you don't want to explicitly set the size, then you can use the method exposed in That article to keep track of the control's size.

**Reply by elsanguchemecatronico** [reply]
Ok, thank you!
